config/exec_resource.go

```diff
@@ -190,12 +195,52 @@ func (e *execResource) enforceState(ctx context.Context) (inDesiredState bool, e
 	case -1:
 		return false, err
 	case 100:
-		return true, nil
+		out, err := execOutput(ctx, e.GetEnforce().GetOutputFilePath())
+		e.enforceOutput = out
+		return true, err
 	default:
 		return false, fmt.Errorf("unexpected return code from enforce: %d, stdout: %s, stderr: %s", code, stdout, stderr)
 	}
 }
 
+func execOutput(ctx context.Context, outputFilePath string) ([]byte, error) {
+	if outputFilePath == "" {
+		return nil, nil
+	}
+
+	clog.Debugf(ctx, "Reading %q for ExecResource output", outputFilePath)
+
+	f, err := os.Open(outputFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("error opening OutputFilePath: %v", err)
+	}
+	defer f.Close()
+
+	// Make a byte slice with a capacity of 1 over maxSize (for simplicity).
+	output := make([]byte, 0, maxExecOutputSize+1)
```

Comment:
Instead of this, can you check if `err == io.EOF` instead? Basically validating that by reading `maxExecOutputSize` bytes, you've either reach the EOF of not?

Reply:
EOF wont happen until the following read, the docs don't explain this clearly but most Read implementations will only return 0, EOF, so for shorter files i would have to make an additional read call to check that we are in fact at the end of the file. I updated the code slightly here to trim that last byte if output is greater.